Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,227 @@
+
+  m_main_layout = new QGridLayout;
+
+  m_main_layout->addWidget(dsp_box, 1, 1, 1, 3);
+  m_main_layout->addWidget(volume_box, 1, 4);
+  m_main_layout->addWidget(backend_box, 2, 1, 1, 4);
+  m_main_layout->addWidget(stretching_box, 3, 1, 1, 4);
+
+  setLayout(m_main_layout);
+}
+
+void AudioPane::ConnectWidgets()
+{
+  connect(m_backend_combo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_volume_slider, &QSlider::valueChanged, [this](int) { SaveSettings(); });
+  connect(m_latency_spin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_stretching_buffer_slider, &QSlider::valueChanged, [this](int) { SaveSettings(); });
+  connect(m_dolby_pro_logic, &QCheckBox::toggled, this, &AudioPane::SaveSettings);
+  connect(m_stretching_enable, &QCheckBox::toggled, this, &AudioPane::SaveSettings);
+  connect(m_dsp_hle, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+  connect(m_dsp_lle, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+  connect(m_dsp_interpreter, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+}
+
+void AudioPane::LoadSettings()
+{
+  auto& settings = Settings::Instance();
+
+  // DSP
+  if (settings.GetDSPHLEEnabled())
+  {
+    m_dsp_hle->setChecked(true);
+  }
+  else
+  {
+    m_dsp_lle->setChecked(settings.GetDSPJITMode());
+    m_dsp_interpreter->setChecked(!settings.GetDSPJITMode());
+  }
+
+  // Backend
+  const auto current = settings.GetCurrentAudioBackend().toStdString();
+  for (const auto& backend : AudioCommon::GetSoundBackends())
+  {
+    m_backend_combo->addItem(tr(backend.c_str()));
+    if (backend == current)
+      m_backend_combo->setCurrentIndex(m_backend_combo->count() - 1);
+  }
+
+  OnBackendChanged();
+
+  // Volume
+  m_volume_slider->setValue(settings.GetVolume());
+  m_volume_label->setText(tr("%1 %").arg(settings.GetVolume()));
+
+  // DPL2
+  m_dolby_pro_logic->setChecked(settings.GetDPLEnabled());
+
+  // Latency
+  m_latency_spin->setValue(settings.GetLatency());
+
+  // Stretch
+  m_stretching_enable->setChecked(settings.GetAudioStretchEnabled());
+  m_stretching_buffer_slider->setValue(settings.GetAudioStretch());
+  m_stretching_buffer_slider->setEnabled(m_stretching_enable->isChecked());
+  m_stretching_buffer_label->setText(tr("%1 ms").arg(settings.GetAudioStretch()));
+}
+
+void AudioPane::SaveSettings()
+{
+  auto& settings = Settings::Instance();
+
+  // DSP
+  settings.SetDSPHLEEnabled(m_dsp_hle->isChecked());
+  settings.SetDSPJITMode(m_dsp_lle->isChecked());
+
+  // Backend
+  const auto selection_localized = m_backend_combo->currentText();
+  const auto backend_localized = tr(settings.GetCurrentAudioBackend().toStdString().c_str());
+
+  if (selection_localized != backend_localized)
+  {
+    for (const auto& backend : AudioCommon::GetSoundBackends())
```

Comment:
It might be easier to store the un-translated strings into the combobox model's user data, instead of working in translated-string-space and doing this repeated translation.